src/data/nat/pairing.lean

```diff
@@ -132,3 +132,35 @@ lemma Union_unpair_prod {α β} {s : ℕ → set α} {t : ℕ → set β} :
 by { rw [← Union_prod], convert surjective_unpair.Union_comp _, refl }
 
 end set
+
+section complete_lattice
+
+lemma supr_unpair_sup {α} [complete_lattice α] {s t : ℕ → α} :
+  (⨆ n : ℕ, s n.unpair.1 ⊔ t n.unpair.2) = (⨆ i : ℕ, s i) ⊔ ⨆ j, t j :=
```

Comment:
Maybe the better statement here is actually
```suggestion
  (⨆ n : ℕ, f n.unpair.1 n.unpair.2) = (⨆ i j: ℕ, f i j) :=
```
which means you only need two versions not four.